src/Compilers/CSharp/Portable/Binder/SimpleLocalScopeBinder.cs

```diff
@@ -21,5 +22,15 @@ protected override ImmutableArray<LocalSymbol> BuildLocals()
         {
             return _locals;
         }
+
+        internal override ImmutableArray<LocalSymbol> GetDeclaredLocalsForScope(CSharpSyntaxNode scopeDesignator)
```

Comment:
I'm a bit surprised you don't store a scope designator on this type, or make it a virtual property, and then test it in this and the next method.